### PR TITLE
[FIX JENKINS-32358] Correctly compute plugin name when multiple sources are passed to install-plugin CLI command

### PR DESCRIPTION
**Pull request by jglick**
I tried

   java -jar jenkins-cli.jar -s … install-plugin -restart …/one.hpi … …/five.hpi

The output looked fine:

    Installing a plugin from local file: …/one.hpi
    …
    Installing a plugin from local file: …/five.hpi

and Jenkins restarted. But the expected plugin versions were not installed. Only `$JENKINS_HOME/plugins/one.jpi` was touched according to file timestamps, and in fact `$JENKINS_HOME/plugins/one/` actually contained the unpacked files from `five.jpi`.

This patch should at least fix that awful bug, though really I think the `-name` argument should be deleted. We deliberately ignore the filename when uploading from the _Advanced_ tab, because it is so often wrong—we parse the manifest instead.

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/cli/InstallPluginCommand.java [hunk 1]
```diff
@@ -75,17 +75,20 @@ protected int run() throws Exception {
         h.checkPermission(PluginManager.UPLOAD_PLUGINS);
         PluginManager pm = h.getPluginManager();
 
+        if (sources.size() > 1 && name != null) {
+            throw new IllegalArgumentException("-name is incompatible with multiple sources");
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/cli/InstallPluginCommand.java]: If so, there should be a bug in JIRA describing the issue and it's impact (if somebody passes such options)

**Review by oleg-nenashev — APPROVED** [review]
Generally LGTM, but IMHO there should be a bug for it in JIRA

**Comment by daniel-beck** [comment]
@oleg-nenashev Let's use https://issues.jenkins-ci.org/browse/JENKINS-32358 for this, close enough.

**Comment by oleg-nenashev** [comment]
Fine for me

**Pull request merged**